### PR TITLE
Lhan/confirm user part3

### PR DESCRIPTION
**Pull request by lhan-the-real**
## Changes made in this PR
Implement ConfirmUser Domain Service layer, and add simple gRPC server function on top of it.

The flow:
1. If user is not an existing user: return error.
2. If the confirmation token is not valid, return confirmationToken not valid.
3. Then, confirm the user by updating the confirmed_at.

Then based on different user account type, do following:
If the user is a receiver:
1. AutoVerifiy user (don't really check verification token, just update the verified_at timestamp)
2. Set default values for the receiver user.
3. return confirmed

If the user is not a receiver(manufacturer user):
1. check if we've ever sent verify request email to pdm account.
2. If sent in the past, then do nothing. 
3. If not sent, generate verification_token, and send email.
4. return verifying

### CODE REVIEW

File: internal/domain/users/domainService.go [hunk 1]
```diff
@@ -164,26 +176,107 @@ func (s *DomainService) ConfirmUser(
...
+	}
+
+	// create verification token, and send verification email to PDM internal email address
+	newVerificationToken, err := s.usersStore.UpdateVerificationToken(ctx, email)
+	if err != nil {
+		return "", pdmerr.NewParent(err, "failed to update verification token")
+	}
+	if !account.ParentID.Valid {
+		return "", pdmerr.New("account is not a parent account")
+	}
```
Comment (sleepyzzzzzz) [inline internal/domain/users/domainService.go:215]: maybe move to the very first part when isReceiver if block just finishes (line196)?

File: internal/domain/users/domainService.go [hunk 2]
```diff
@@ -164,26 +176,107 @@ func (s *DomainService) ConfirmUser(
...
+	parent, err := s.parentsStore.GetParent(ctx, account.ParentID.Int32)
+	if err != nil {
+		return "", pdmerr.NewParent(err, "couldn't find parent by parent ID")
+	}
```
Comment (sleepyzzzzzz) [inline internal/domain/users/domainService.go:219]: also this part

File: internal/domain/users/domainService.go [hunk 3]
```diff
@@ -164,26 +176,107 @@ func (s *DomainService) ConfirmUser(
...
+
 		return "#{Constant::ENV_TO_TOOL[Padrino.env]}/receiver_confirmed", nil
 	}
 
-	// if is not receiver, then try to verify parent user
-	// user email domain is valid in parent_domains?
-	// the parent domain's parent id == given account's parentID?
-	// if so, update the user's verified at.
+	// if the account is a manufacturer account, validate verification token
+	verificationToken, err := s.usersStore.GetVerificationToken(ctx, email)
```
Comment (sleepyzzzzzz) [inline internal/domain/users/domainService.go:198]: why not just use userInfo.VerificationToken?

**Review by sleepyzzzzzz — COMMENTED** [review]
everything looks great, just fewer comments

File: internal/domain/users/domainService.go (hunk 1 above)
Comment (lhan-the-real) [reply internal/domain/users/domainService.go:215]: Yes! Thanks!

File: internal/domain/users/domainService.go (hunk 2 above)
Comment (lhan-the-real) [reply internal/domain/users/domainService.go:219]: Yep! did it!

File: internal/domain/users/domainService.go (hunk 3 above)
Comment (lhan-the-real) [reply internal/domain/users/domainService.go:207]: Really good point! I made the change. This is much cleaner!

**Pull request merged**